python/hail/expr/functions.py

```diff
@@ -515,6 +515,21 @@ def fisher_exact_test(c1, c2, c3, c4):
     return _func("fet", ret_type, c1, c2, c3, c4)
 
 
+@typecheck(x=oneof(expr_float32, expr_float64))
+def floor(x):
```

Comment:
why don't these accept integer types and do implicit conversion?
```
In [3]: np.floor(2)
Out[3]: 2.0
```

Reply:
I think there's pretty much nowhere we should accept floats and not ints.

Reply:
Doesn't it seem like a mistake to call floor on an integer? I can only think of things like floor'ing the return value of an if, but in that case the int will be promoted by the `if` before the `floor` sees it.

Reply:
I don't think anything in numpy that expects a float would error on int. Shouldn't we do the same?